### DISCUSSION TITLE
Event descriptors list and Triggers

### DISCUSSION THREAD

**Discussion by larry104** [alexzhornyak/ScxmlEditor-Tutorial#109] [General]
It seems if you have a list of event desciptors e.g.

<transition event="key1Pressed key2Pressed key3Pressed" target="StateShape21"/> 

and you define a content trigger they both get converted into a single trigger instead of one trigger for each event name.  Is there a workaround by predefining the content triggers?

**Comment by alexzhornyak** [comment]
Hi, when all triggers have the same params, they will be treated as single

![image](https://user-images.githubusercontent.com/18611095/227139752-5b7951f7-0bdf-4c1d-b9c1-3ffdf244557e.png)

If you need separate triggers with the same event name, modify anything in params, for example set Max to 1, 2, 3

![image](https://user-images.githubusercontent.com/18611095/227140348-d8a3ef7e-aeae-4521-9b27-54829fc05b58.png)

or change appearance

![image](https://user-images.githubusercontent.com/18611095/227140671-0ba9ae7a-fa4b-4cca-9747-3542b9ea0393.png)

**Comment by larry104** [comment]
Thanks for the answer! In my case the triggers were different like key1Pressed, key2PPressed, etc. Is there a solution for that as well?

**Comment by alexzhornyak** [comment]
Share some screenshots, what you have at the moment and what do you want to achieve

**Comment by larry104** [comment]
Ideally the left and right should create the same triggers. IMO a combined trigger does not make sence since events are processed in a event queue so I assume there is no way that all 3 could get triggered simultaneously.

Even better would be that you could write key$Pressed and set min=1 and max=3 and it would be translated to the correct scxml statment
`<transition event="key1Pressed key2Pressed key3Pressed" target="ShapeState2"/>`

![scxml](https://user-images.githubusercontent.com/2013720/227827116-9da622d4-294b-4d0c-936f-3493702fd042.JPG)

**Comment by alexzhornyak** [comment]
We have special element called **Trigger** for handling such things

![image](https://user-images.githubusercontent.com/18611095/228250347-2cae8a33-2c0c-4dd4-b01f-d6a98452710a.png)

https://user-images.githubusercontent.com/18611095/228252566-e71e4857-3f93-4a4a-9cb4-cc60736b83e1.mp4

P.S. But I would like also to show how I use such things in Scxml

https://user-images.githubusercontent.com/18611095/228254779-a3542900-60b1-4ef4-ace0-9767006bf213.mp4

![image](https://user-images.githubusercontent.com/18611095/228255071-d80d8696-6587-4ad6-89f9-39c12a3bfb04.png)

**Comment by larry104** [comment]
Thanks a lot for the detailed explanation. I like that you can globally define triggers that is very useful. I admit your way of doing it is much more elegant!!!
